apps/cowswap-frontend/src/common/pure/OrderProgressBarV2/index.tsx

```diff
@@ -355,79 +495,101 @@ function FinishedStep({
-              </>
-            )}
-          </styledEl.Surplus>
+          <styledEl.FinishedImageContent>
+            <styledEl.FinishedTagLine>
+              {shouldShowSurplus ? (
+                <styledEl.BenefitSurplusContainer ref={surplusContainerRef}>
+                  <span ref={surplusTextRef} style={{ fontSize: `${surplusFontSize}px` }}>
+                    I just received surplus on my
+                    <styledEl.TokenPairTitle
+                      title={order ? `${order.inputToken.symbol}/${order.outputToken.symbol}` : 'N/A'}
+                    >
+                      {order
+                        ? truncateWithEllipsis(`${order.inputToken.symbol}/${order.outputToken.symbol}`, 30)
+                        : 'N/A'}
+                    </styledEl.TokenPairTitle>{' '}
+                    trade
+                    <styledEl.Surplus
+                      ref={surplusRef}
+                      showSurplus={!!shouldShowSurplus}
+                      style={{ fontSize: `${surplusSize}px` }}
+                      data-content={
+                        shouldShowSurplus && surplusPercentValue !== 'N/A' ? `+${surplusPercentValue}%` : 'N/A'
+                      }
+                    >
+                      <span>
+                        {shouldShowSurplus && surplusPercentValue !== 'N/A' ? `+${surplusPercentValue}%` : 'N/A'}
+                      </span>
+                    </styledEl.Surplus>
+                  </span>
+                </styledEl.BenefitSurplusContainer>
+              ) : (
+                <styledEl.BenefitSurplusContainer ref={benefitContainerRef}>
+                  <styledEl.BenefitText ref={benefitTextRef} fontSize={benefitFontSize}>
+                    {randomBenefit}
+                  </styledEl.BenefitText>
+                </styledEl.BenefitSurplusContainer>
+              )}
+            </styledEl.FinishedTagLine>
+
+            <styledEl.FinishedLogo>
+              <ProductLogo
+                variant={ProductVariant.CowSwap}
+                theme="light"
+                overrideColor={UI.COLOR_PRIMARY_DARKER}
+                height={19}
+                logoIconOnly
+              />
+              <b>CoW Swap</b>
+            </styledEl.FinishedLogo>
+          </styledEl.FinishedImageContent>
         </styledEl.ProgressImageWrapper>
       </styledEl.ProgressTopSection>
 
       <styledEl.ConclusionContent>
-        <styledEl.TransactionStatus>
-          <PiCheckCircleFill />
+        <styledEl.TransactionStatus flexFlow="column" margin={'0 auto 14px'}>
+          <Lottie
+            animationData={theme.darkMode ? LOTTIE_GREEN_CHECKMARK_DARK : LOTTIE_GREEN_CHECKMARK}
+            loop={false}
+            autoplay
+            style={{ width: '56px', height: '56px' }}
+          />
           Transaction completed!
         </styledEl.TransactionStatus>
 
-        {solvers?.length && (
+        {solvers && solvers.length > 0 && (
```

Comment:
Means the same thing. If `solvers` is falsy or `length` is 0, it won't get into this clause.
Did the linter/IDE complain?

Reply:
Don't recall, but updated now